R/infrastructure.R

```diff
@@ -291,6 +291,27 @@ use_package_doc <- function(pkg = ".") {
   writeLines(out, file.path(pkg$path, path))
 }
 
+# help function for use_package
+# takes the same arguments as use_package
+build_package_txt <- function(package, version = NULL,
+                              compare =  c(">=", ">", "==", "<=", "<")) {
```

Comment:
I can tell you that where I used to work, there was serious consideration for `==` in our private package development since we didn't want to have to update our code when dependencies changed. In the end, we said "what the hell" and went with `>=`. That being said, changes to `stringr` (from 0.6.3 to 1.0.0) ended up "forcing" some time to package maintenance.  

I don't agree with that we should drop `compare` because it's "unlikely" that someone will use the other options. I thought the spirit of `devtools` is to help people build packages (with minimal opinions, beyond best practice). Restricting the user to either `package` or `package (>= packageVersion)` in the `DESCRIPTION` file seems mighty opinionated.